wolfcrypt/src/asn.c

```diff
@@ -160,63 +142,75 @@
 
 #elif defined(TIME_OVERRIDES)
     /* user would like to override time() and gmtime() functionality */
-
     #ifndef HAVE_TIME_T_TYPE
-        typedef long time_t;
+        #define USE_WOLF_TIME_T
     #endif
-    extern time_t XTIME(time_t * timer);
-
     #ifndef HAVE_TM_TYPE
-        struct tm {
-            int  tm_sec;     /* seconds after the minute [0-60] */
-            int  tm_min;     /* minutes after the hour [0-59] */
-            int  tm_hour;    /* hours since midnight [0-23] */
-            int  tm_mday;    /* day of the month [1-31] */
-            int  tm_mon;     /* months since January [0-11] */
-            int  tm_year;    /* years since 1900 */
-            int  tm_wday;    /* days since Sunday [0-6] */
-            int  tm_yday;    /* days since January 1 [0-365] */
-            int  tm_isdst;   /* Daylight Savings Time flag */
-            long tm_gmtoff;  /* offset from CUT in seconds */
-            char *tm_zone;   /* timezone abbreviation */
-        };
+        #define USE_WOLF_TM
     #endif
-    extern struct tm* XGMTIME(const time_t* timer, struct tm* tmp);
 
-    #ifndef HAVE_VALIDATE_DATE
-        #define XVALIDATE_DATE(d, f, t) ValidateDate((d), (f), (t))
-    #endif
+    /* forward declarations */
+    extern time_t XTIME(time_t * timer);
```

Comment:
Same as above. The typedef time_t is known because I added "USE_WOLF_TIME_T" at new line 131, which does the "typedef long time_t" below at new line 208.